### DISCUSSION TITLE
Fast refresh messes up state of component with useForm()

### DISCUSSION THREAD

**Discussion by gip** [vercel/next.js#17850] [Help]
# Bug report

## Describe the bug

I've got a complex component using `useForm()` (from `react-hook-form`). Note that this component does have child components that update the state too. Works well, including in prod. However fast refresh is messing up the state. After fast refresh, (most) fields are not available when `handleSubmit()` is called. 

## To Reproduce

Happy to try to repro is useful - however, it's a complex component and I'd rather save the time if it's a known issue or a feature.

## Expected behavior

Internal state of the form should remain the same.

## Screenshots

N/A

## System information

- OS: macOS
- Browser (if applies); Chrome
- Version of Next.js: 9.5.5
- Version of Node.js: 14.4

## Additional context

Could be related to https://github.com/vercel/next.js/issues/13268
And being able to simply disable fast refresh would be welcome.

**Comment by jamesmosier** [comment]
It's hard to say what might be causing this without a reproduction. Could you create one with the minimal amount of code required to reproduce this?

There are some [limitations](https://nextjs.org/docs/basic-features/fast-refresh#limitations) of Fast refresh as well to keep in mind.
